opm/input/eclipse/Schedule/ResCoup/GrupSlav.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+  Copyright 2024 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <opm/input/eclipse/Schedule/ResCoup/GrupSlav.hpp>
+#include <opm/input/eclipse/Schedule/ResCoup/ReservoirCouplingInfo.hpp>
+#include <opm/input/eclipse/Schedule/Schedule.hpp>
+#include <opm/input/eclipse/Schedule/ScheduleState.hpp>
+#include <opm/input/eclipse/Schedule/ScheduleStatic.hpp>
+#include <opm/input/eclipse/Parser/ParserKeywords/G.hpp>
+#include <opm/common/OpmLog/OpmLog.hpp>
+#include <opm/common/utility/OpmInputError.hpp>
+#include "../HandlerContext.hpp"
+
+#include <fmt/format.h>
+
+#include <limits>
+#include <stdexcept>
+
+namespace Opm {
+namespace ReservoirCoupling {
+GrupSlav GrupSlav::serializationTestObject()
+{
+    return GrupSlav{"MANI-D", "D1-M", FilterFlag::MAST, FilterFlag::MAST, FilterFlag::MAST,
+                    FilterFlag::MAST, FilterFlag::MAST, FilterFlag::MAST, FilterFlag::MAST};
+}
+
+bool GrupSlav::operator==(const GrupSlav& rhs) const {
+    return
+        this->m_name == rhs.m_name &&
+        this->m_master_group_name == rhs.m_master_group_name &&
+        this->m_oil_prod_flag == rhs.m_oil_prod_flag &&
+        this->m_liquid_prod_flag == rhs.m_liquid_prod_flag &&
+        this->m_gas_prod_flag == rhs.m_gas_prod_flag &&
+        this->m_fluid_volume_prod_flag == rhs.m_fluid_volume_prod_flag &&
+        this->m_oil_inj_flag == rhs.m_oil_inj_flag &&
+        this->m_water_inj_flag == rhs.m_water_inj_flag &&
+        this->m_gas_inj_flag == rhs.m_gas_inj_flag;
+}
+
+// NOTE: This function is needed by Boost.Test, see test/ReservoirCoupling/GrupSlavTest.cpp
+std::ostream& operator<<(std::ostream& os, const GrupSlav::FilterFlag& flag) {
+    switch (flag) {
+        case GrupSlav::FilterFlag::MAST:
+            os << "MAST";
+            break;
+        case GrupSlav::FilterFlag::SLAV:
+            os << "SLAV";
+            break;
+        case GrupSlav::FilterFlag::BOTH:
+            os << "BOTH";
+            break;
+        default:
+            throw std::invalid_argument("Invalid filter flag");
+    }
+    return os;
+}
+
+} // namespace ReservoirCoupling
+
+void checkValidSlaveGroupName(const std::string& name, HandlerContext& handlerContext)
+{
+    auto rescoup = handlerContext.state().rescoup();
```

Comment:
IS there a reason for the copy or would it be better to use `const auto&`?

Reply:
Thanks for reviewing this. Good catch, I do not see why would need a copy here either. I will update the PR to use `const auto&` instead